cluster-autoscaler/expander/nodecountbalance/nodecountbalance.go

```diff
@@ -0,0 +1,44 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodecountbalance
+
+import (
+	"k8s.io/autoscaler/cluster-autoscaler/expander"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+type nodecountbalance struct {
+}
+
+// NewStrategy returns an expansion strategy that randomly picks between node groups
+func NewStrategy() expander.Strategy {
+	return &nodecountbalance{}
+}
+
+// BestOption selects from the expansion options to balance node groups in their node counts
+func (r *nodecountbalance) BestOption(expansionOptions []expander.Option, nodeInfo map[string]*schedulercache.NodeInfo) *expander.Option {
+	pos := 0
+	for i, o := range expansionOptions {
+		if i == 0 {
+			continue
+		}
+		if o.NodeCount < expansionOptions[pos].NodeCount {
```

Comment:
The problem I'm trying to address is balancing nodes between node groups with identical instance type in different zones. An example use case could look like this: user has node groups with 2 different instance types - 2 cpu and 32 cpu. He has a node group in 2 different zones for each instance type for HA (so 4 node groups in total).

In this case I don't see any rationale for balancing the number of 2-cpu machines with 32-cpu machines, regardless if we balance by node count (1:1 nodes) or by capacity (1:16 nodes). I want to balance the number of nodes with given instance type between 2 zones, preferably not limiting user's choice of strategy for choosing which instance type he wants (using leastwaste, mostpods, or his custom code). From my point of view there are 2 separate decisions to be made: "what kind of node do I want?" and "which node group providing such nodes should I choose?".

I think that for the purpose of solving the problem I'm describing it's better to let expander.Strategy focus on answering the first question and have some simple external bookkeeping for balancing the node group sizes between zones (@mwielgus idea of 'node group family' is one way to do it). OTOH I don't think any specific expander.Strategy suits my use-case, as I want to still allow user to choose any strategy they want (or implement their own). 

I think the issue you're trying to solve is a subproblem of the one I described (there is just one instance type and it needs to be balanced between zones). If I'm correct I don't think we need two separate solutions, as your problem can be solved by putting all your node groups into a single 'node group family'. Of course the alternative is that I misunderstood your use-case :)